app/services/layer-groups.js

```diff
@@ -1,107 +1,136 @@
 import Service from '@ember/service';
-import { A } from '@ember/array';
-import { copy } from 'ember-copy';
+import { set } from '@ember/object';
 
-const areIncomingLayerGroupsDefault = function(layerGroupsToShow, layerGroupsByDefault) {
-  const layersToShowAreDefault = layerGroupsToShow.every(
-    (layerGroupToShow, idx) => layerGroupToShow === layerGroupsByDefault[idx],
-  );
+export const isEqual = function(array1, array2) {
+  array1.sort();
+  array2.sort();
 
-  return layersToShowAreDefault;
+  return array1.every((value, index) => value === array2[index]);
+};
+
+const matchesIdentifierFor = function(layerGroup) {
+  return param => (param.id || param) === layerGroup.id;
```

Comment:
Where is param coming from?

Reply:
Oy! It's confusing. it's the param to an array function (which has no parentheses). This func returns a function - (is this "currying"?).

Reply:
In this case it is passe to an array method, param comes from typical iteratable